File: __algos/algos-0.0.1/ts/src/data_structures/queue/__tests__/q_test.ts
import {assertEquals, assert} from "~/dev_deps.ts"
import {Queue} from "../q.ts"

Deno.test("Enqueue", () => {
  const queue = new Queue<number>()
  queue.enqueue(1)
  queue.enqueue(2)
  queue.enqueue(3)
  assertEquals(queue.print(), [1, 2, 3])
})

Deno.test("Dequeue", () => {
  const queue = new Queue<number>()
  queue.enqueue(1)
  queue.enqueue(2)
  queue.enqueue(3)
  assertEquals(queue.dequeue(), 1)
  assertEquals(queue.print(), [2, 3])

  assertEquals(queue.dequeue(), 2)
  assertEquals(queue.print(), [3])

  assertEquals(queue.dequeue(), 3)
  assertEquals(queue.print(), [])
  assert(queue.isEmpty())
})

Deno.test("Peek", () => {
  const queue = new Queue<number>()
  queue.enqueue(1)
  queue.enqueue(2)
  queue.enqueue(3)
  assertEquals(queue.peek(), 1)
  assertEquals(queue.print(), [1, 2, 3])
})

File: algorithms-data-structures/go/src/data-structures/queue/q_test.go
package queue

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestQ(t *testing.T) {
	q := Queue{}
	q.Enqueue(1)
	q.Enqueue(2)
	q.Enqueue(3)

	assert.Equal(t, 3, q.Len())
	assert.Equal(t, "1,2,3", q.Print())

	node := q.Dequeue()
	assert.Equal(t, 1, node.Value)
	assert.Equal(t, 2, q.Len())

	node = q.Peek()
	assert.Equal(t, 2, node.Value)

}
